lib/connection.js

```diff
@@ -449,6 +449,11 @@ Connection.prototype.openUri = function(uri, options, callback) {
       delete options.useCreateIndex;
     }
 
+    if ('useFindAndModify' in options) {
+      this.base.options.useFindAndModify = !!options.useFindAndModify;
```

Comment:
I like the idea of these options, but this is a big no-no. Connections should not modify the mongoose global's options, because you're in for a nasty surprise if you create multiple connections with different useFindAndModify options:

```javascript
mongoose.createConnection(uri1, { useFindAndModify: true });
mongoose.createConnection(uri2, {}); // Forgot to set useFindAndModify

mongoose.options.useFindAndModify; // false!
```

Instead, please use the same pattern as `useCreateIndex`: if schema option is set, use that. If schema option is not set, use the connection option if it is set. If not, fall back to the mongoose global's option.

Reply:
Now it differences the query, the connection level and the global.